### ISSUE TITLE
How to install on MacOS

### ISSUE THREAD

**Issue by JVAQUEROM** [pdfarranger/pdfarranger#279]
Hi there,

I like pdfarranger and I recommend it to some friends using MacOS (I don't use that OS myself). However, they do not really know how to install it, and the repository has instructions for other OS but not for MacOS. So If there is a chance someone provides instructions for MacOS it would be great.

Thank you and keep up the good work!

**Comment by jeromerobert** [comment]
As far as I know no one has ever tried. I'm not a MacOS user. I told my colleagues (they are Apple fanboys) about pdfarranger and they said that the MacOS default PDF viewer already has the pdfarranger features. That would explain the lack of interest for MacOS installation instructions.

Googling... Some says to install Python GTK3 with Brew or MacPort or Conda. I expect that having a Quartz GTK3 with Python support will be the most difficult step. Building QPDF and PikePDF should be easier.

I guess that none of the regular contributors will work on that, so I add the "help needed tag".

I would be nice if a MacOS user could tell us if this incident is really of interest or not.

**Comment by dreua** [comment]
> I guess that none of the regular contributors will work on that, so I add the "help needed tag".

I did download and install (in VMs) some linux distros over the years to reproduce and fix some pdfarranger or packaging bugs, but with MacOS it's not that simple. Even if it were it would probably be too much work and too little interest on my side to make that work. 

I'd say we close this as "Wontfix" unless an interested mac developer turns up.

**Comment by dreua** [comment]
I'll close this since there seems to be no more interest in this topic and we can't really do anything here right now. Everyone please feel free to add comments if you want to express interest in this topic or if you want to work on it.

**Comment by Endle** [comment]
I'm a both mac and Fedora user, and I'm interest in using pdfarranger under mac, as I could have a similar experience across various platforms.

I'm optimistic with migrating it to mac, as the major dependencies of pdfarranger worked well on mac (mupdf, pikepdf)

Currently I have two blockers (there may be more)
1. libintl.so.8 is required by pdfarranger. I could skip the i18n part as a workaround.
2. I need `pip install PyGObject` for `import gi`. I had some dependency issue, but PyGObject supports mac (https://pygobject.readthedocs.io/en/latest/getting_started.html#macosx-getting-started), so I think this is a solvable problem.

I'll post my updates here, but I can't give any promises yet.

**Comment by jeromerobert** [comment]
You should replace libintl.so.8 in source code by the libintl provided by your gettext installation

See:
* https://formulae.brew.sh/formula/gettext
* https://github.com/pdfarranger/pdfarranger/blob/7b73781/pdfarranger/pdfarranger.py#L76

**Comment by Endle** [comment]
@jeromerobert sorry for pinging you. Do you know if there is a CLI of pdfarranger?

I could get it run now, but it would crash if I save the edited PDF. With CLI, I can debug that if it's due to UI (gtk) or underlying issues

**Comment by jeromerobert** [comment]
If you do that https://github.com/pdfarranger/pdfarranger?tab=readme-ov-file#for-developers from a terminal you should be able to see all error messages. Except if it's a segfault in the native part (GTK or Poppler or qpdf) the errors should be pretty clear. If not you may post them here and we'll try to help.

**Comment by Endle** [comment]
I've got pdfarranger to run on my mac. Easier than I had expected.

There are two tasks needs to be done
1) libintl. Currently I just turned off translation. I'll replace my hack with some proper code to execute
2) GTK schema is required to run FileChooser. A mac user can override `GSETTINGS_SCHEMA_DIR` to provide the schema files. One can also execute `gsettings list-recursively` to see if the schema files are loaded properly.

I'll send my PR later

**Comment by Endle** [comment]
Created my PR https://github.com/pdfarranger/pdfarranger/pull/1045/files

**Comment by dreua** [comment]
With that PR merged I believe that can be closed again. Some more QA and potentially useful infromation landed in this PR but the most important information is now in the Readme. Thanks again!

**Comment by rudeboyo** [comment]
Hi, I don't know if this is the right place to be barking up this tree, but I'm currently working with a publishing team that uses pdfarranger and I've been trying to figure out how to install on macos. I followed this thread as well as possible with no knowledge on GTK3.

I've downloaded the pdf-arranger-main.zip and have been parsing through the files, but I don't know where I would implement the instructions to install/run pdfinstaller on Mac. There's two major problems I have as a layman.

1. There's no exectuable x.app file (like there would be an .exe file in the windows package) so I don't actually know where to start or what I'd enter into command line if I were to use that. 
2. How do I create $GSETTINGS_SCHEMA_DIR/org.gtk.Settings.FileChooser.gschema.xml and $DYLD_LIBRARY_PATH/libintl.8.dylib files and what folders would I put them into?

**Comment by dreua** [comment]
I'm not a mac user so hopefully someone else can help here. In the meantime, have you read all comments in the PR https://github.com/pdfarranger/pdfarranger/pull/1045#issuecomment-2290073631? 

It would be great to have an easier way to run PDF Arranger on mac but that really needs to come from someone actually using a mac, I''m happy to help from the Linux/PDF Arranger side.

**Comment by rudeboyo** [comment]
when I run the Homebrew & Pip dependencies suggested by Endle I get

'pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1'

also

'Did not find pkg-config by name 'pkg-config'
            Found pkg-config: NO
            Did not find CMake 'cmake'
            Found CMake: NO
            Run-time dependency cairo found: NO (tried framework)'

something in the subprocess isn't working, I'm guessing there's supposed to be a pkg-config file? but I'm not sure what that should be or if there's just a file that's named wrong?

**Comment by Endle** [comment]
@rudeboyo Have you tried the command in https://github.com/pdfarranger/pdfarranger/pull/1045#issuecomment-2290073631 ?

```
git clone https://github.com/Endle/pdfarranger.git
cd pdfarranger; git checkout mac
brew install  pygobject3 gtk+3 poppler gettext
ln -s /opt/homebrew/lib/libintl.8.dylib .
pip3 install --user pikepdf PyGObject python-dateutil img2pdf
python3 -m pdfarranger
```


Please let me know if it didn't work, and I'll try to reproduce this issue at my end






> 'pip subprocess to install build dependencies did not run successfully. │ exit code: 1'
>

Can you share the command you executed, and upload the full terminal output to gist?

> also
> 
> 'Did not find pkg-config by name 'pkg-config' Found pkg-config: NO Did not find CMake 'cmake' Found CMake: NO Run-time dependency cairo found: NO (tried framework)'
> 

How can I get this terminal output?

**Comment by rudeboyo** [comment]
Yes that's the original command I'm working with per dreua's suggestion.
I'm using homebrew through the standard mac terminal.

This is the full terminal output on gist, I just recreated it again using the same command:
https://gist.github.com/rudeboyo/3d8d210a153024cb3f3181bfd955f0d5.js

**Comment by dreua** [comment]
The `brew` command already fails, do you have brew installed? I can't tell you how to do that but there are probably good instructions online.

@Endle I think this should now work just as fine with the current PDF Arranger release, no need to use your fork's mac branch any more, right?

Apple does make it especially hard for us to distribute software for their platform unfortunately. (Even running a VM on foreign hardware is against TOS :angry:  )

**Comment by Endle** [comment]
```
fatal: destination path 'pdfarranger' already exists and is not an empty directory.
error: pathspec 'mac' did not match any file(s) known to git
zsh: command not found: brew
```

The brew failed at very early stage.

> I think this should now work just as fine with the current PDF Arranger release, no need to use your fork's mac branch any more, right?


Yes, just use the main branch.

> Apple does make it especially hard for us to distribute software for their platform unfortunately.

Sadly, true. I don't use homebrew, maybe someone could pack pdfarranger to homebrew. https://github.com/orgs/Homebrew/discussions/533


I'm a nix user and I'm trying to add pdfarranger to nixpkgs. https://github.com/NixOS/nixpkgs/pull/349832

**Comment by dreua** [comment]
Now that the NixOS package supports mac (thank you!), would that be an easy and recommended way for the average mac user to install PDF Arranger on macos? In that case, I'd love to have some instructions in the readme (PRs welcome).
